jerry-ext/handler/handler-assert.c

```diff
@@ -132,3 +186,234 @@ jerryx_handler_assert (const jerry_value_t func_obj_val, /**< function object */
+} /* jerryx_handler_assert_not_equal */
+
+/**
+ * Handler for the assert not strict equal operation.
+ *
+ * @return true - if the two arguments are not strict equal.
+ *         otherwise - this function calls jerryx_fatal.
+ *         Note that in this case the function does not return.
+ */
+static jerry_value_t
+jerryx_handler_assert_not_strict_equal (const jerry_value_t func_obj_val, /**< function object */
+                                        const jerry_value_t this_p, /**< this arg */
+                                        const jerry_value_t args_p[], /**< function arguments */
+                                        const jerry_length_t args_cnt) /**< number of function arguments */
+{
+  return jerryx_handler_assert_equal_helper (func_obj_val, this_p, args_p, args_cnt, JERRYX_OP_NOT_STRICT_EQUAL);
+} /* jerryx_handler_assert_not_strict_equal */
+
+/**
+ * Handler to cause an assertion failure.
+ */
+static jerry_value_t
+jerryx_handler_assert_fail (const jerry_value_t func_obj_val, /**< function object */
+                            const jerry_value_t this_p, /**< this arg */
+                            const jerry_value_t args_p[], /**< function arguments */
+                            const jerry_length_t args_cnt) /**< number of function arguments */
+{
+  (void) func_obj_val; /* unused */
+  (void) this_p; /* unused */
+
+  if (args_cnt > 0)
+  {
+    jerryx_fatal (args_p[0]);
+  }
+
+  jerryx_fatal (jerry_create_null ());
+  return jerry_create_null ();
+} /* jerryx_handler_assert_fail */
+
+/**
+ * Handler function to check if a function throws an error.
+ */
+static jerry_value_t
+jerryx_handler_assert_throws (const jerry_value_t func_obj_val, /**< function object */
+                              const jerry_value_t this_p, /**< this arg */
+                              const jerry_value_t args_p[], /**< function arguments */
+                              const jerry_length_t args_cnt) /**< number of function arguments */
+{
+  (void) func_obj_val; /* unused */
+  (void) this_p; /* unused */
+
+  if (args_cnt > 0 && jerry_value_is_function (args_p[0]))
+  {
+    jerry_value_t value = jerry_call_function (args_p[0], jerry_create_undefined (), NULL, 0);
+
+    if (jerry_value_is_error (value))
+    {
+      jerry_value_t result = jerry_get_value_from_error (value, true);
+      if (args_cnt > 1 && jerry_value_is_undefined (args_p[1]))
+      {
+        jerry_release_value (result);
+        return jerry_create_boolean (true);
+      }
+
+      jerry_binary_operation_t operation;
+
+      if (!jerry_value_is_object (args_p[1]))
+      {
+        operation = JERRY_BIN_OP_STRICT_EQUAL;
+      }
+      else
+      {
+        operation = JERRY_BIN_OP_INSTANCEOF;
+      }
+
+      jerry_value_t comparison = jerry_binary_operation (operation, result, args_p[1]);
+      if (jerry_get_boolean_value (comparison))
+      {
+        jerry_release_value (result);
```

Comment:
After the second argument of `jerry_get_value_from_error` is changed to false, we will need to release `value` here instead.